examples-api-use/pixel-example.cc

```diff
@@ -0,0 +1,150 @@
+// -*- mode: c++; c-basic-offset: 2; indent-tabs-mode: nil; -*-
+// Small example how to use the input bits
+//
+// This code is public domain
+// (but note, that the led-matrix library this depends on is GPL v2)
+
+#include "led-matrix.h"
+#include "graphics.h"
+
+#include <getopt.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <signal.h>
+#include <termios.h>
+
+char getch() {
+        char buf = 0;
+        struct termios old;
+        if (tcgetattr(0, &old) < 0)
+                perror("tcsetattr()");
+        old.c_lflag &= ~ICANON;
+        old.c_lflag &= ~ECHO;
+        old.c_cc[VMIN] = 1;
+        old.c_cc[VTIME] = 0;
+        if (tcsetattr(0, TCSANOW, &old) < 0)
+                perror("tcsetattr ICANON");
+        if (read(0, &buf, 1) < 0)
+                perror ("read()");
+        old.c_lflag |= ICANON;
+        old.c_lflag |= ECHO;
+        if (tcsetattr(0, TCSADRAIN, &old) < 0)
+                perror ("tcsetattr ~ICANON");
+        return (buf);
+}
+
+using namespace rgb_matrix;
+
+volatile bool interrupt_received = false;
+static void InterruptHandler(int signo) {
+  interrupt_received = true;
```

Comment:
Need flag? Could make canvas pointer global and clean up here. (Null checks)

Reply:
No, you can't. You must not do anything complex in a signal handler.

Reply:
Yeah these are not interrupt handles. Granted this suck pretty bad. In all fairness this is probably not needed at all.